src/Paraunit/File/TempDirectory.php

```diff
@@ -39,15 +39,26 @@ public function getTempDirForThisExecution()
 
     /**
      * @return string
+     *
+     * @throws \RuntimeException
      */
     public static function getTempBaseDir()
     {
-        foreach (self::$tempDirs as $directory) {
+        $dirs = self::$tempDirs;
+        // Fallback to sys temp dir
+        $dirs[] = sys_get_temp_dir();
+
+        foreach ($dirs as $directory) {
             if (file_exists($directory)) {
                 $baseDir = $directory . DIRECTORY_SEPARATOR . 'paraunit';
-                self::mkdirIfNotExists($baseDir);
 
-                return $baseDir;
+                try {
+                    self::mkdirIfNotExists($baseDir);
+
+                    return $baseDir;
+                } catch (\RuntimeException $e) {
+                    // ignore and try next dir
+                }
             }
         }
```

Comment:
Last change needed: you should throw the same exception here, since if you reach here, it means that all attempts failed.